stage1/init/common/pod.go

```diff
@@ -487,7 +487,9 @@ func writeShutdownService(p *stage1commontypes.Pod) error {
 	shutdownVerb := "exit"
 	// systemd <v227 doesn't allow the "exit" verb when running as PID 1, so
 	// use "halt".
-	if systemdVersion < 227 {
+	// If systemdVersion is 0 it means it couldn't be guessed, assume it's new
+	// enough for "systemctl exit".
```

Comment:
You can say somewhere (commit message or code) that it happens when building with:
```
./configure --with-stage1-flavors=src --with-stage1-systemd-version=master
```